packages/remix-eslint-config/rules/typescript.js

```diff
@@ -1,54 +1,49 @@
-const OFF = 0;
-const WARN = 1;
-const ERROR = 2;
-
 module.exports = {
-  "no-dupe-class-members": OFF,
-  "no-undef": OFF,
-
-  // Add TypeScript specific rules (and turn off ESLint equivalents)
-  "@typescript-eslint/consistent-type-assertions": WARN,
-  "@typescript-eslint/consistent-type-imports": WARN,
-
-  "no-array-constructor": OFF,
-  "@typescript-eslint/no-array-constructor": WARN,
+  // TODO: These rules might be nice to enable... we should investigate eventually!
+  "@typescript-eslint/ban-ts-comment": "off",
+  "@typescript-eslint/ban-types": "off",
+  "@typescript-eslint/no-empty-function": "off",
+  "@typescript-eslint/no-empty-interface": "off",
+  "@typescript-eslint/no-explicit-any": "off",
+  "@typescript-eslint/no-inferrable-types": "off",
+  "@typescript-eslint/no-namespace": "off",
+  "@typescript-eslint/no-non-null-assertion": "off",
+  "@typescript-eslint/no-var-requires": "off",
+  "no-var": "off",
+  "prefer-rest-params": "off",
 
-  // There is a bug w/ @typescript-eslint/no-duplicate-imports triggered
-  // by multiple imports inside of module declarations. We should reenable
-  // this rule when the bug is fixed.
-  // https://github.com/typescript-eslint/typescript-eslint/issues/3071
-  "no-duplicate-imports": OFF,
-  // "@typescript-eslint/no-duplicate-imports": WARN,
-
-  "no-redeclare": OFF,
-  "@typescript-eslint/no-redeclare": ERROR,
-  "no-use-before-define": OFF,
+  // These rules are nice and we want to configure over the defaults
   "@typescript-eslint/no-use-before-define": [
-    WARN,
+    "error",
```

Comment:
Regarding `"warn"` vs `"error"`: I generally recommend against using warnings except for transitionary code (enabling a new rule with many existing violations). https://github.com/eslint/eslint/issues/16696 -> https://github.com/eslint/eslint/pull/16882 ->  https://eslint.org/docs/latest/use/configure/rules#rule-severities. In typescript-eslint@v5, most preset config rule severities are set to error; in v6, _all_ will be error.

Reply:
We have a workflow that runs `npx prettier -w . && npx eslint . --fix` & commits the changes, so I guess that's why the team decided to put these as `warn` instead of `error` 🤔

https://github.com/remix-run/remix/blob/main/.github/workflows/format.yml

Reply:
It doesn't look like this or the below ones can be autofixed - so error may be correct.  Might fail a few CI builds on updating remix but if it's for a more accurate codebase then folks can always override the rule back down to `warn`.